commit e03187e12cae57c09b521b6f7dd7c7f9aa2b62e9
Author: Anshuman Gupta <anshuman.gupta@intel.com>
Date:   Mon Jan 11 13:41:15 2021 +0530

    drm/i915/hdcp: MST streams support in hdcp port_data
    
    Add support for multiple mst stream in hdcp port data
    which will be used by RepeaterAuthStreamManage msg and
    HDCP 2.2 security f/w for m' validation.
    
    Security f/w doesn't have any provision to mark the stream_type
    for each stream separately, it just take single input of
    stream_type while authenticating the port and applies the
    same stream_type to all streams. So driver mark each stream_type
    with common highest supported content type for all streams
    in DP MST Topology.
    
    Security f/w supports RepeaterAuthStreamManage msg and m'
    validation only once during port authentication and encryption.
    Though it is not compulsory, security fw should support dynamic
    update of content_type and should support RepeaterAuthStreamManage
    msg and m' validation whenever required.
    
    v2:
    - Init the hdcp port data k for HDMI/DP SST stream.
    v3:
    - Cosmetic changes. [Uma]
    v4:
    - 's/port_auth/hdcp_port_auth'. [Ram]
    - Commit log improvement.
    v5:
    - Comment and commit log improvement. [Ram]
    v6:
    - Check first connector connected status before intel_encoder_is_mst
      to avoid any NULL pointer dereference.
    
    Cc: Ramalingam C <ramalingam.c@intel.com>
    Reviewed-by: Uma Shankar <uma.shankar@intel.com>
    Reviewed-by: Ramalingam C <ramalingam.c@intel.com>
    Tested-by: Karthik B S <karthik.b.s@intel.com>
    Signed-off-by: Anshuman Gupta <anshuman.gupta@intel.com>
    Tested-by: Juston Li <juston.li@intel.com>
    Acked-by: Juston Li <juston.li@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20210111081120.28417-15-anshuman.gupta@intel.com

diff --git a/drivers/gpu/drm/i915/display/intel_display_types.h b/drivers/gpu/drm/i915/display/intel_display_types.h
index 63f94e2bd23f..d2c4df4ee8b9 100644
--- a/drivers/gpu/drm/i915/display/intel_display_types.h
+++ b/drivers/gpu/drm/i915/display/intel_display_types.h
@@ -1516,10 +1516,12 @@ struct intel_digital_port {
 	enum phy_fia tc_phy_fia;
 	u8 tc_phy_fia_idx;
 
-	/* protects num_hdcp_streams reference count, hdcp_port_data */
+	/* protects num_hdcp_streams reference count, hdcp_port_data and hdcp_auth_status */
 	struct mutex hdcp_mutex;
 	/* the number of pipes using HDCP signalling out of this port */
 	unsigned int num_hdcp_streams;
+	/* port HDCP auth status */
+	bool hdcp_auth_status;
 	/* HDCP port data need to pass to security f/w */
 	struct hdcp_port_data hdcp_port_data;